r/tests/testthat/test-dplyr-query.R

```diff
@@ -293,3 +293,23 @@ test_that("No duplicate field names are allowed in an arrow_dplyr_query", {
     )
   )
 })
+
+test_that("show_exec_plan()", {
+  expect_snapshot(
+    tbl %>%
+      arrow_table() %>%
+      filter(dbl > 2, chr != "e") %>%
+      select(chr, int, lgl) %>%
+      mutate(int_plus_ten = int + 10) %>%
+      show_exec_plan()
+  )
+
+  expect_snapshot(
+    tbl %>%
+      record_batch() %>%
+      filter(dbl > 2, chr != "e") %>%
+      select(chr, int, lgl) %>%
+      mutate(int_plus_ten = int + 10) %>%
+      show_exec_plan()
+  )
+})
```

Comment:
It would be nice to have tests for the following as well (they might be somewhere overkill, see the paragraph below this, but those go through enough other machinery in R that it would be good to confirm that they don't fail):

* a `summarise()`
* a join (it could even be a self join)
* a dataset querying multiple files (this one might be better in the datasets test file since the setup is already done there...)

You also might consider asserting specific limited things about the output but not the whole output in a snapshot (e.g. does it include the columns referenced, does it have the boiler plate we expect won't change like `ExecPlan .* nodes`). The content of the output is more or less directly from C++, so if that were to change we would need to update these tests even if the C++ <-> R code hasn't changed at all.

Reply:
Done